app/core/src/main/java/com/fsck/k9/controller/Preconditions.kt

```diff
@@ -4,6 +4,10 @@ package com.fsck.k9.controller
 
 import com.fsck.k9.K9
 
+fun <T : Any> checkNotNull(value: T?) {
+    requireNotNull(value)
```

Comment:
`requireNotNull()` throws `IllegalArgumentException`. `checkNotNull()` throws `IllegalStateException`. Our methods should probably follow the same convention.

That being said. The methods in `MessagingController` and `MessagingControllerCommands` should probably use `requireNotNull()` instead of `checkNotNull()`.